jetty-io/src/main/java/org/eclipse/jetty/io/ByteBufferPool.java

```diff
@@ -158,7 +158,7 @@ public void release(ByteBuffer buffer)
         private final int _capacity;
         private final int _maxSize;
         private final AtomicInteger _size;
-        private long _lastUpdate = System.nanoTime();
+        private volatile long _lastUpdate = System.nanoTime();
```

Comment:
In this way we always pay the cost of `volatile`, which is updated much more often than it's being read.

Frankly, I would leave it non-`volatile`.
Next best is using `AtomicLong.lazySet()` in 9.4.x and `VarHandle` in 10.0.x.
I would refrain using `volatile` unconditionally like proposed here, if possible.

Reply:
You're right that we could do better than simply using `volatile`.

On JDK 8, `AtomicLong`'s `lazySet()`/`get()` is probably best.
On JDK 9+, `lazySet()`/`getOpaque()` provide the exact semantics we want at the cheapest possible cost.

Let me update this PR, and I'll include the `get()` -> `getOpaque()` change during the 10.0.x merge.